ddtrace/span.py

```diff
@@ -275,6 +277,9 @@ def set_tag(self, key, value=None):
 
         # Key should explicitly be converted to a float if needed
         elif key in NUMERIC_TAGS:
+            if value is None:
```

Comment:
Before, having value be None would result in a warning/debug log before. Now it seems like we just silently fail to set the tag?

Reply:
Right this would no longer result in a debug line for those cases of `NUMERIC_TAGS` which ends up being only the analytics sample rate. In that case we were passing along the `None` value and logging an error, which is why we would see this in logs:

```
DEBUG:ddtrace.span:ignoring not number metric _dd1.sr.eausr:None
```

Reply:
Is there a reason to change this behaviour? Seems like it'd still be useful to have the log

Reply:
While this old behavior is incorrect, I'll keep the behavior as is in this PR and pull out that change into another PR.